### PR TITLE
Misc test improvements

### PR DESCRIPTION
**Pull request by marcelocantos**

### CODE REVIEW

File: syntax/test_helpers.go [hunk 1]
```diff
@@ -59,19 +59,16 @@ func AssertCodeEvalsToType(t *testing.T, expected interface{}, code string) bool
 	return true
 }
 
-// AssertCodePanics asserts that the code panics when executed.
+// AssertCodePanics asserts that the exprs evaluate to the same value.
```
Comment (dancantos) [inline syntax/test_helpers.go]: I think the old comment was more accurate

File: syntax/test_helpers.go [hunk 2]
```diff
@@ -59,19 +59,16 @@ func AssertCodeEvalsToType(t *testing.T, expected interface{}, code string) bool
...
-		return false
-	}
-	return true
+	return assert.Panics(t, func() {
+		pc := ParseContext{SourceDir: ".."}
+		ast, err := pc.Parse(parser.NewScanner(code))
+		if assert.NoError(t, err, "parsing code: %s", code) {
+			codeExpr := pc.CompileExpr(ast)
+			codeExpr.Eval(rel.EmptyScope) //nolint:errcheck
+		}
```
Comment (nofun97) [inline syntax/test_helpers.go]: you can use `require.NoError` instead.

Comment (nofun97) [reply syntax/test_helpers.go]: to replace `assert.NoError`

Comment (marcelocantos) [reply syntax/test_helpers.go]: That fails the entire test. I want to allow the test to continue after this assertion fails.

**Pull request merged**